app/connectors/HeaderGenerator.scala

```diff
@@ -20,14 +20,20 @@ import config.AppConfig
 import play.api.http.HeaderNames._
 import uk.gov.hmrc.http.HeaderCarrier
 import utils.Logger.logger
+import utils.UUIDGenerator
 
 import javax.inject.{Inject, Singleton}
 
 @Singleton
-class HeaderGenerator @Inject()(appConfig: AppConfig) {
+class HeaderGenerator @Inject()(appConfig: AppConfig, idGenerator: UUIDGenerator) {
 
   def headersForPEGA()(implicit hc: HeaderCarrier): Seq[(String, String)] = {
-    val headers = Seq("Environment" -> appConfig.pegaEnvironment)
+    val headers = Seq(
+      "CorrelationId" -> idGenerator.generateUUID,
+      "Environment" -> appConfig.pegaEnvironment
+    )
+    val correlationId = headers.find(_._1 == "CorrelationId").map(_._2).getOrElse("")
```

Comment:
Above line 31 could you not define `val correlationId = idGenerator.generateUUID` and then you can reference `correlationId` in `headers` and in the logging?